core/src/autogluon/core/utils/utils.py

```diff
@@ -21,86 +21,89 @@
 from ..constants import BINARY, REGRESSION, MULTICLASS, SOFTCLASS, QUANTILE
 from ..metrics import accuracy, root_mean_squared_error, pinball_loss, Scorer
 
-logger = logging.getLogger(__name__)
-
 
+logger = logging.getLogger(__name__)
 
-# TODO: RespirceManager to replace these two hacks
-class _get_cpu_count:
-    def __init__(self):
-        self.mock = None
 
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
+class ResourceManager():
+    """Manager that fetches system related info"""
+    
+    @staticmethod
+    def get_cpu_count():
         return multiprocessing.cpu_count()
     
-
-class _get_gpu_count_all:
-    """
-    Attempts to get number of GPUs available for use via multiple means.
-    """
-    # FIXME: update to use only torch for TIMM or find a better GPU detection strategy
-    # FIXME: get_gpu_count by itself doesn't always work for Windows
-    def __init__(self):
-        self.mock = None
-
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
-        num_gpus = _get_gpu_count_cuda()
+    @staticmethod
+    def get_gpu_count_all():
+        num_gpus = ResourceManager._get_gpu_count_cuda()
         if num_gpus == 0:
-            num_gpus = get_gpu_count_mxnet()
+            num_gpus = ResourceManager.get_gpu_count_mxnet()
             if num_gpus == 0:
-                num_gpus = get_gpu_count_torch()
+                num_gpus = ResourceManager.get_gpu_count_torch()
```

Comment:
Tiny edit: We should actually be calling `get_gpu_count_torch` before we call `get_gpu_count_mxnet` now that Torch is an expected dependency and MXNet is being deprecated. This lowers the risk of MXNet import occurring when it isn't actually being used, which could cause issues.

Reply:
Yea, I was thinking the same. Let me update it

Reply:
@Innixma be aware of this thing https://github.com/awslabs/autogluon/issues/2042

Reply:
Oh good point @gradientsky . Then we can keep the import order as is for now, but add a code comment about this specific issue so we are aware of it when viewing the code.